src/poetry/plugins/application_plugin.py

```diff
@@ -24,4 +24,6 @@ def commands(self) -> list[type[Command]]:
     def activate(self, application: Application) -> None:
         for command in self.commands:
             assert command.name is not None
-            application.command_loader.register_factory(command.name, lambda: command())
+            application.command_loader.register_factory(
+                command.name, lambda: command()  # noqa: B023
```

Comment:
This one can, I think, be considered as a false positive, since we do want late binding here.